sklearn/utils/_sparse.py

```diff
@@ -0,0 +1,62 @@
+"""Control sparse interface based on config"""
+
+# Authors: The scikit-learn developers
+# SPDX-License-Identifier: BSD-3-Clause
+
+import scipy as sp
+
+from .._config import get_config
+
+
+def _as_sparse(X_sparse):
```

Comment:
Do you think it'll be simpler to have `_as_sparse` be `_select_interface_if_sparse`?

Reply:
I guess the difference between function is whether we know already that the input is sparse. We usually know whether it is sparse. But there are some cases where the input could be dense or sparse.

Looking at this again now, the cost of a simpler one-function approach is small. We always check `issparse` so we could forego the exception and just pass-through anything that is not sparse.

Do you have a suggestion for the name of the single function? `_as_sparse` suggests it converts everything to sparse. But `_select_interface_if_sparse` is a long name. :)  Maybe `_align_api_if_sparse`?

Reply:
I'm okay with `_align_api_if_sparse`.